changelogs/unreleased/gh-7898-msgpack-object-indexation.md

```diff
@@ -0,0 +1,4 @@
+## feature/lua/msgpack
+
+* Added `msgpack.object:index` method which performs indexation similar to
```

Comment:
> It'd also be nice to have the fields auto-completed in console:

Basically, AFAI, `__autocomplete` is equivalent to `__serialize` which, in turn, is equivalent to `decode` — wouldn't that be too expensive?

Reply:
> AFAI, `__autocomplete` is equivalent to `__serialize` which, in turn, is equivalent to `decode`

No, we just need to iterate over all keys in the map.

> wouldn't that be too expensive?

It's okay, because it's only used in console. Autocompleting MsgPack fields may be optional, but we do need to autocomplete keys from the object metatable, because this works now:

```
tarantool> o = require('msgpack').object({})
---
...

tarantool> o.
o.decode(    o.iterator(
```

Reply:
Added `__autocomplete` metamethod.